giraph-core/src/main/java/org/apache/giraph/edge/ByteArrayEdges.java

```diff
@@ -94,6 +94,9 @@ public void add(Edge<I, E> edge) {
     } catch (IOException e) {
       throw new IllegalStateException("add: Failed to write to the new " +
           "byte array");
+    } catch (NegativeArraySizeException negativeArraySizeException) {
+      throw new IllegalStateException("add: Too many edges for a vertex, " +
```

Comment:
Could we add the ID of the vertex in the exception message? It may be useful for debugging purposes.